src/proof_of_possession.rs

```diff
@@ -170,17 +175,20 @@ impl ProofOfPossession {
         Ok(jws::encode_sign_custom_header(&payload, jwk, &header)?)
     }
 
-    pub async fn from_proof(
-        proof: &Proof,
-        resolver: &dyn DIDResolver,
-    ) -> Result<Self, ParsingError> {
+    pub async fn from_proof<R>(proof: &Proof, resolver: R) -> Result<Self, ParsingError>
```

Comment:
Instead of having a type parameter `R`, you can simply write `resolver: impl JWKResolver`, that may be more readable.